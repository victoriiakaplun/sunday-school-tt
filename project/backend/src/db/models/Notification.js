const db = require('@src/db');
const { DataTypes } = require('sequelize');

const Notification = db.define(
  'Notification',
  {
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
      allowNull: false,
    },
    type: {
      type: DataTypes.ENUM('CREATED', 'CONFIRMED', 'REJECTED', 'DELETED'),
      allowNull: false,
    },
    isRead: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
    },
  },
  {
    freezeTableName: true,
  },
);

module.exports = Notification;
