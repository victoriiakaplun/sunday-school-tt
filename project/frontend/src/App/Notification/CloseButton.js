import React from 'react';

function CloseButton({ onClose }) {
  return (
    <button type="button" className="delete" onClick={onClose}>
      x
    </button>
  );
}

export default CloseButton;
