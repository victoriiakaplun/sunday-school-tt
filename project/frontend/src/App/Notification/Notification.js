import React, { useEffect } from 'react';
import styled from '@emotion/styled';
import Button from '../../components/Button';

function Notification({ onClose, delay, children }) {
  useEffect(() => {
    setTimeout(() => onClose(), 5000);
  }, [onClose, delay]);
  return (
    <Container>
      <div className="notification is-warning">
        <button type="button" className="delete" onClick={onClose}>
          x
        </button>
        {children}
      </div>
    </Container>
  );
}

const Container = styled.div`
  width: 250px;
  text-align: center;
  position: absolute;
  right: 0;
  bottom: 0;
`;

Notification.defaultProps = {
  delay: 5,
};
export default Notification;
