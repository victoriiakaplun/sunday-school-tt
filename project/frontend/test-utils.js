import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

initFontAwesomeLibrary();

// import '@testing-library/jest-dom/extend-expect';

// re-export everything
export * from '@testing-library/react';
