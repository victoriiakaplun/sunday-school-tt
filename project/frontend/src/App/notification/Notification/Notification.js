import React, { useEffect } from 'react';
import classNames from 'classnames';
import CloseButton from '../../../components/button/CloseButton';

import styles from './Notification.scss';

function Notification({ onClose, delay, type, children }) {
  useEffect(() => {
    setTimeout(() => onClose(), delay);
  }, [onClose, delay]);

  const containerClasses = classNames(styles.container, 'notification', `is-${type}`);

  return (
    <div className={containerClasses}>
      <CloseButton onClose={onClose} />
      {children}
    </div>
  );
}

Notification.defaultProps = {
  delay: 10000,
};

export default Notification;
