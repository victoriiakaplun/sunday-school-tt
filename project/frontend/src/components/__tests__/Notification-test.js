import { render } from 'test-utils';

import Notification from '../../App/notification/Notification';

describe('Notification', () => {
  it('renders correctly', () => {
    const { container } = render(<Notification>Notification</Notification>);
    expect(container.firstChild).toMatchSnapshot(`
      <div
       class="notification is-warning">
      Notification
      </div>
    `);
  });
});
