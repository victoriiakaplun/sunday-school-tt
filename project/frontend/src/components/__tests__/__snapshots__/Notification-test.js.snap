// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly: 
      <div
       class="notification is-warning">
      Notification
      </div>
     1`] = `
<div
  class="css-1n4a782-Container e19f66hx0"
>
  <div
    class="notification is-undefined"
  >
    <button
      class="delete"
      type="button"
    >
      x
    </button>
    Notification
  </div>
</div>
`;
